addons/base_import_module/models/ir_module.py

```diff
@@ -207,7 +214,10 @@ def import_zipfile(self, module_file, force=False):
                             terp = ast.literal_eval(f.read().decode())
                     except Exception:
                         continue
-                    for filename in terp.get('data', []) + terp.get('init_xml', []) + terp.get('update_xml', []):
+                    files_to_import = terp.get('data', []) + terp.get('init_xml', []) + terp.get('update_xml', [])
+                    if with_demo:
```

Comment:
shouldn't we check if the database is in demo mode?

Reply:
No, because it should work on database that have no demo to only display the ones of the zip module.

Reply:
the runbot tests the module over a database with demo data installed